processor/processor.go

```diff
@@ -797,6 +798,106 @@ func (proc *HandleT) updateSourceEventStatsDetailed(event types.SingularEventT,
 		}
 	}
 }
+
+type TransformationStatusT struct {
+	sourceID          string
+	destID            string
+	destination       *backendconfig.DestinationT
+	eventsToTransform []transformer.TransformerEventT
+	eventsByMessageID map[string]types.SingularEventWithReceivedAt
+	response          *transformer.ResponseT
+	uniqueMessageIds  map[string]struct{}
+}
+
+func getEvent(singularEvent types.SingularEventT, receivedAt time.Time) map[string]interface{} {
```

Comment:
make this return value also struct instead of map, since keys will be same always ?